### PR TITLE
fix: Show Not Enough Pool Liquidity on Huge Swaps

### PR DESCRIPTION
**Pull request by JoseRFelix**
Addresses #1260

### CODE REVIEW

**Comment by sunnya97** [comment]
Task linked: [CU-860pp6kn5 USDC USDT max swap](https://app.clickup.com/t/860pp6kn5)

File: packages/web/localizations/en.json [hunk 1]
```diff
@@ -371,6 +371,8 @@
     "invalidSlippage": "Invalid slippage",
     "noSendCurrency": "Currency to send not set",
     "insufficientBal": "Insufficient balance",
+    "insufficientLiquidity": "Insufficient liquidity",
+    "noRoute": "No Route for this trade",
```
Comment (jonator) [inline packages/web/localizations/en.json:375]: Actually can we please not capitalize "Route"

File: packages/stores/src/ui-config/trade-token-in-config.ts [hunk 2]
```diff
@@ -410,6 +437,9 @@ export class ObservableTradeTokenInConfig extends AmountConfig {
     }
 
     if (this.amount) {
+      if (this._notEnoughLiquidity) return new NotEnoughLiquidityError();
```
Comment (jonator) [inline packages/stores/src/ui-config/trade-token-in-config.ts:440]: I think the no balance error should take precedence over the liquidity error. I see other DEXs do that, and IMO it's better optics and more accurate.

**Review by jonator — CHANGES_REQUESTED** [review]

**Comment by jonator** [comment]
Ehh I'm not really a fan of the UX of creating branches from ClickUp, the names are illegible. We should focus on linking to issues and secondarily linking to ClickUp (from ClickUp).

File: packages/stores/src/ui-config/trade-token-in-config.ts [hunk 3]
```diff
@@ -410,6 +437,9 @@ export class ObservableTradeTokenInConfig extends AmountConfig {
     }
 
     if (this.amount) {
+      if (this._error instanceof NoRouteError) return this._error;
+      if (this._notEnoughLiquidity) return new NotEnoughLiquidityError();
+
```
Comment (jonator) [inline packages/stores/src/ui-config/trade-token-in-config.ts:442]: I think this needs to be moved below the balance code below (below that last if statement).

**Review by jonator — CHANGES_REQUESTED** [review]

File: packages/stores/src/ui-config/trade-token-in-config.ts (hunk 3 above)
Comment (JoseRFelix) [reply packages/stores/src/ui-config/trade-token-in-config.ts:442]: If I moved it down there, `Insufficient Balance` would take precedence. Users who like to experiment can get confused when the `out` token stops providing estimates, just like it happened to Jim. I think leaving it like it's right now will benefit both users. What do you think?

Comment (jonator) [reply packages/stores/src/ui-config/trade-token-in-config.ts:442]: I guess that's true given that we don't calculate past the pool's liquidity as other DEXs do. However, it's not ideal to be advertising that we don't have enough liquidity. Both are true to some extent.

SGTM

**Review by jonator — APPROVED** [review]

**Pull request merged**